core/src/main/scala/org/apache/spark/rdd/CoalescedRDD.scala

```diff
@@ -58,7 +58,7 @@ private[spark] case class CoalescedRDDPartition(
       val parentPreferredLocations = rdd.context.getPreferredLocs(rdd, p.index).map(_.host)
       preferredLocation.exists(parentPreferredLocations.contains)
     }
-    if (parents.size == 0) 0.0 else (loc.toDouble / parents.size.toDouble)
```

Comment:
I realize the irony in continuing the thread, and this is not specifically directed at this PR alone: 

No, it's quite easy to merge and our test process is automated. (My assertion here though is that this is not faster -- it calls size == 0 -- even though I prefer it for readability too.) 

It's that thousands of people are on the mailing list, and hundreds will spend 10 seconds reading this, and a few committers will spend minutes or more at least, and it adds up to hundreds of dollars of people's time, if you want to think of it that way. (Remember the project gets 20-30 new issues a day, too, and committers only have part of each day to spend on these things.) Your time is better spent looking into open starter JIRAs, if we're going to ask the community to invest even this small amount of time.

Same observation goes for people asking on user@ -- you're asking for many hours of time of attention from people across the world every time in aggregate. Is your question worth $1000? It may literally cost that. Alternatively, think of how valuable a good answer is in time saved!